### ISSUE TITLE
Bug: Loading sharable config with multiple globs ignores the last glob.

### ISSUE THREAD

**Issue by daniel-zh23** [eslint/eslint#19115]
### Environment

Node version: 20.18.0
npm version: 10.8.2
Local ESLint version: 9.13.0
Global ESLint version: N/A
Operating System: Windows 10 Enterprise 22H2 build: 19045.5011

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```js
import globalConfig from "./ESLint/eslint.config.mjs";

export default [
    {
        ignores: [""]
    },
    ...globalConfig
];
```
</details>

So I am currently migrating from eslintrc to flat config. We have a global submodule for general rules that we have and use in different projects. Using that config when linting .ts files I get an error and not the warning that I am supposed to get. Overall all the rules that I have in that second glob are ignored. Using the eslint inspector when filtering by *.ts shows it correctly, but after lint it's not.

```js
import eslint from '@eslint/js';
import tsEslint from 'typescript-eslint';
import globals from "globals";
import typescriptES from "@typescript-eslint/eslint-plugin";
import stylisticTS from "@stylistic/eslint-plugin";

export default [
  eslint.configs.recommended,
  ...tsEslint.configs.recommended,
  {
    languageOptions: {
      ecmaVersion: "latest",
      sourceType: "module",
      parser: tsEslint.parser,
      globals: {
        ...globals.browser,
        ...globals.jquery,
        ...globals.es2021
      }
    },
    plugins: {
      "@typescript-eslint": typescriptES,
      "@stylistic/ts": stylisticTS
    },
    rules: {
        "no-unused-vars": "error", //Disallows unused variables
    },
  },
  {
    files: ["*.ts", "*.tsx"],
    rules: {
        "no-unused-vars": "off",
    },
  },
];

```

### What did you expect to happen?

Looking at the documentation I expected the latest rule at the bottom glob to overwrite the one that's in the higher glob, but only for *.ts and *.tsx files, while being an error for different file like *.js.

### What actually happened?

The error rule is being applied to the *.ts files.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiaW1wb3J0IGVzbGludCBmcm9tICdAZXNsaW50L2pzJztcbmltcG9ydCB0c0VzbGludCBmcm9tICd0eXBlc2NyaXB0LWVzbGludCc7XG5pbXBvcnQgZ2xvYmFscyBmcm9tIFwiZ2xvYmFsc1wiO1xuaW1wb3J0IHR5cGVzY3JpcHRFUyBmcm9tIFwiQHR5cGVzY3JpcHQtZXNsaW50L2VzbGludC1wbHVnaW5cIjtcbmltcG9ydCBzdHlsaXN0aWNUUyBmcm9tIFwiQHN0eWxpc3RpYy9lc2xpbnQtcGx1Z2luXCI7XG5cbmV4cG9ydCBkZWZhdWx0IFtcbiAgZXNsaW50LmNvbmZpZ3MucmVjb21tZW5kZWQsXG4gIC4uLnRzRXNsaW50LmNvbmZpZ3MucmVjb21tZW5kZWQsXG4gIHtcbiAgICBsYW5ndWFnZU9wdGlvbnM6IHtcbiAgICAgIGVjbWFWZXJzaW9uOiBcImxhdGVzdFwiLFxuICAgICAgc291cmNlVHlwZTogXCJtb2R1bGVcIixcbiAgICAgIHBhcnNlcjogdHNFc2xpbnQucGFyc2VyLFxuICAgICAgZ2xvYmFsczoge1xuICAgICAgICAuLi5nbG9iYWxzLmJyb3dzZXIsXG4gICAgICAgIC4uLmdsb2JhbHMuanF1ZXJ5LFxuICAgICAgICAuLi5nbG9iYWxzLmVzMjAyMVxuICAgICAgfVxuICAgIH0sXG4gICAgcGx1Z2luczoge1xuICAgICAgXCJAdHlwZXNjcmlwdC1lc2xpbnRcIjogdHlwZXNjcmlwdEVTLFxuICAgICAgXCJAc3R5bGlzdGljL3RzXCI6IHN0eWxpc3RpY1RTXG4gICAgfSxcbiAgICBydWxlczoge1xuICAgICAgICBcIm5vLXVudXNlZC12YXJzXCI6IFwiZXJyb3JcIiwgLy9EaXNhbGxvd3MgdW51c2VkIHZhcmlhYmxlc1xuICAgIH0sXG4gIH0sXG4gIHtcbiAgICBmaWxlczogW1wiKi50c1wiLCBcIioudHN4XCJdLFxuICAgIHJ1bGVzOiB7XG4gICAgICAgIFwibm8tdW51c2VkLXZhcnNcIjogXCJvZmZcIixcbiAgICB9LFxuICB9LFxuXTtcbiIsIm9wdGlvbnMiOnsicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOlsiZXJyb3IiXSwiZm9yLWRpcmVjdGlvbiI6WyJlcnJvciJdLCJnZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOlsiZXJyb3IiXSwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOlsiZXJyb3IiXSwibm8tY2xhc3MtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbXBhcmUtbmVnLXplcm8iOlsiZXJyb3IiXSwibm8tY29uZC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29uc3QtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0YW50LWJpbmFyeS1leHByZXNzaW9uIjpbImVycm9yIl0sIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6WyJlcnJvciJdLCJuby1jb250cm9sLXJlZ2V4IjpbImVycm9yIl0sIm5vLWRlYnVnZ2VyIjpbImVycm9yIl0sIm5vLWRlbGV0ZS12YXIiOlsiZXJyb3IiXSwibm8tZHVwZS1hcmdzIjpbImVycm9yIl0sIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6WyJlcnJvciJdLCJuby1kdXBlLWVsc2UtaWYiOlsiZXJyb3IiXSwibm8tZHVwZS1rZXlzIjpbImVycm9yIl0sIm5vLWR1cGxpY2F0ZS1jYXNlIjpbImVycm9yIl0sIm5vLWVtcHR5IjpbImVycm9yIl0sIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1lbXB0eS1wYXR0ZXJuIjpbImVycm9yIl0sIm5vLWVtcHR5LXN0YXRpYy1ibG9jayI6WyJlcnJvciJdLCJuby1leC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjpbImVycm9yIl0sIm5vLWZhbGx0aHJvdWdoIjpbImVycm9yIl0sIm5vLWZ1bmMtYXNzaWduIjpbImVycm9yIl0sIm5vLWdsb2JhbC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW1wb3J0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1pbnZhbGlkLXJlZ2V4cCI6WyJlcnJvciJdLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6WyJlcnJvciJdLCJuby1sb3NzLW9mLXByZWNpc2lvbiI6WyJlcnJvciJdLCJuby1taXNsZWFkaW5nLWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1uZXctbmF0aXZlLW5vbmNvbnN0cnVjdG9yIjpbImVycm9yIl0sIm5vLW5vbm9jdGFsLWRlY2ltYWwtZXNjYXBlIjpbImVycm9yIl0sIm5vLW9iai1jYWxscyI6WyJlcnJvciJdLCJuby1vY3RhbCI6WyJlcnJvciJdLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOlsiZXJyb3IiXSwibm8tcmVkZWNsYXJlIjpbImVycm9yIl0sIm5vLXJlZ2V4LXNwYWNlcyI6WyJlcnJvciJdLCJuby1zZWxmLWFzc2lnbiI6WyJlcnJvciJdLCJuby1zZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjpbImVycm9yIl0sIm5vLXNwYXJzZS1hcnJheXMiOlsiZXJyb3IiXSwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOlsiZXJyb3IiXSwibm8tdW5kZWYiOlsiZXJyb3IiXSwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOlsiZXJyb3IiXSwibm8tdW5yZWFjaGFibGUiOlsiZXJyb3IiXSwibm8tdW5zYWZlLWZpbmFsbHkiOlsiZXJyb3IiXSwibm8tdW5zYWZlLW5lZ2F0aW9uIjpbImVycm9yIl0sIm5vLXVuc2FmZS1vcHRpb25hbC1jaGFpbmluZyI6WyJlcnJvciJdLCJuby11bnVzZWQtbGFiZWxzIjpbImVycm9yIl0sIm5vLXVudXNlZC1wcml2YXRlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tdW51c2VkLXZhcnMiOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1iYWNrcmVmZXJlbmNlIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtY2F0Y2giOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1lc2NhcGUiOlsiZXJyb3IiXSwibm8td2l0aCI6WyJlcnJvciJdLCJyZXF1aXJlLXlpZWxkIjpbImVycm9yIl0sInVzZS1pc25hbiI6WyJlcnJvciJdLCJ2YWxpZC10eXBlb2YiOlsiZXJyb3IiXX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

This is from the eslint inspector.
![Image](https://github.com/user-attachments/assets/b8d28fe0-b202-40f5-8312-0a81874fee7b)

**Comment by mdjermanovic** [comment]
Hi @daniel-zh23, thanks for the issue!

Can you please provide a StackBlitz or GitHub repro repo as it isn't really possible to reproduce this in the ESLint playground?

> ```files: ["*.ts", "*.tsx"]```

Note that these patterns match only files in the root. Updating to `files: ["**/*.ts", "**/*.tsx"]` might solve the problem.

**Comment by daniel-zh23** [comment]
Omg. That was it. Maybe after 9 hours of migrating is a bit too much.
Thank you.
